### DISCUSSION TITLE
lab4-1跑 source run_sim時發生錯誤

### DISCUSSION THREAD

**Discussion by sungsungwu** [bol-edu/HLS-SOC-Discussions#125] [Q&A]
請問跑 source run_sim時會出現以下錯誤該如何解決呢?
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/113113868/b7b0f90f-1efd-421f-a9df-02befe7990d3)

還有講義內這三行code也執行過了，但卻會產生以下錯誤，請問是因為這個問題才造成上面的錯誤嗎?
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/113113868/0116cbb7-0c58-4803-9549-b1c8c25ba697)
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/113113868/b8faec11-babd-4360-a6ae-ac0989a46451)
感謝!

**Comment by LingChiYang** [comment]
同學好

看起來你的C code沒辦法compile，我想terminal有告訴你錯哪裡了。若還是找不到，再請同學貼上沒辦法compile的那段code。

**Reply by weixiang1118** [reply]
可是lab4-1的c code 只能改fir.c這份檔案，終端機報錯的卻是其他的c code

**Reply by LingChiYang** [reply]
看起來是同學的fir.c和其他寫好的function在link時有衝突。Compiler並沒有那麼聰明，有時是不會把全部錯誤的地方都顯示出。
若同學不方便貼出自己的fir.c，不如先請同學試試看簡單的function能不能compile。

**Reply by weixiang1118** [reply]
好的 謝謝你 我再試試看!

**Reply by ruei7916** [reply]
你們的fir.c是不是有`#include <stub.c>`? 如果有的話把它拿掉試試看

**Reply by weixiang1118** [reply]
對! 後來發現是include的問題，已解決，謝謝!!
